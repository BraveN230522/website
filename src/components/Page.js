import React from 'react';
import Head from 'next/head';

import { UserReducerProvider } from '@utils/userContext';
import { CartProvider } from '@utils/cartContext';

import { useAuthentication } from '@components/Authentication/useAuthentication';
import AuthenticationWidget from '@components/Authentication/AuthenticationWidget';
import Modal from '@components/common/Modal';

import { useRouter } from 'next/router';
import { getPageMetaForRoute } from '@utils/routes';
import Footer from './Layout/footer';

const Page = ({ title, description, children }) => {
  const { shouldRequireAuthenticatedAccess, verifyAuthenticationControl } = useAuthentication();

  const router = useRouter();
  const pageMeta = getPageMetaForRoute(router);

  return (
    <UserReducerProvider>
      <CartProvider>
        <Head>
          <meta name="viewport" content="width=device-width, user-scalable=no" />
          <meta name="description" content={pageMeta ? pageMeta.description : description} />
          <meta name="facebook-domain-verification" content="khzvesqfvsftqebvwhczwor912fixu" />
          <title>{pageMeta ? pageMeta.title : title}</title>
        </Head>

        {children}

        <Footer />

        <Modal isInitiallyOpened={shouldRequireAuthenticatedAccess} onClosing={verifyAuthenticationControl}>
          {onClose => <AuthenticationWidget closeWidget={onClose} />}
        </Modal>
      </CartProvider>
    </UserReducerProvider>
  );
};

Page.defaultProps = {
  title: 'Sports Tickets & Fan Travel Tips | Sports Where I Am',
  description:
    'Sports Where I Am is a sport travel service offering fully guaranteed tickets & fan travel tips across  65,000 sporting events across 240+ cities world-wide',
};

export default Page;
