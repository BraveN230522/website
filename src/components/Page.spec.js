import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';

import Page from './Page';

describe('<Page />', () => {
  it('renders Page layout unchanged', () => {
    const { container } = render(<Page />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });
});
