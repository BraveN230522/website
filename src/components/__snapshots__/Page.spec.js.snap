// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders Page layout unchanged 1`] = `
.emotion-100 {
  background: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 45px 22px 26px;
}

.emotion-98 {
  max-width: 1255px;
  width: 100%;
  margin: 0 auto;
}

.emotion-98 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:60em) {
  .emotion-98 > div {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

.emotion-88 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 70%;
}

@media (max-width:60em) {
  .emotion-88 {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 100%;
  }
}

@media (max-width:48em) {
  .emotion-88 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    padding: 20px;
    text-align: center;
  }
}

.emotion-20 {
  list-style-type: none;
  padding: 0;
}

.emotion-20:nth-of-type(1) {
  width: 15%;
}

.emotion-20:nth-of-type(2) {
  width: 17%;
}

.emotion-20:nth-of-type(3) {
  width: 23%;
}

.emotion-20:nth-of-type(4) {
  width: 20%;
}

@media (max-width:48em) {
  .emotion-20:first-of-type {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
    border-bottom: 1px solid #272727;
    width: 50%;
  }

  .emotion-20:nth-of-type(2) {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
    border-bottom: 1px solid #272727;
    width: 50%;
  }

  .emotion-20:nth-of-type(3) {
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
    width: 50%;
  }

  .emotion-20:nth-of-type(4) {
    -webkit-order: 4;
    -ms-flex-order: 4;
    order: 4;
    width: 50%;
  }
}

.emotion-0 {
  color: #ffffff;
  display: block;
  font-family: Helvetica,sans-serif;
  font-size: 17px;
  font-weight: 400;
  line-height: 25px;
  margin-bottom: 16px;
}

@media (max-width:60em) {
  .emotion-0 {
    font-size: 18px;
    line-height: 40px;
    opacity: 0.7;
  }

  .emotion-0:hover {
    opacity: 1;
  }
}

.emotion-94 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 70%;
}

@media (max-width:60em) {
  .emotion-94 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    width: 100%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 0.5em 0 1.5em;
  }
}

.emotion-90 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 22px;
}

@media (max-width:60em) {
  .emotion-90 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-92 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-92 svg {
  margin: 0 10px;
  -webkit-transition: fill ease-out 0.3s;
  transition: fill ease-out 0.3s;
}

.emotion-92 svg:hover {
  fill: #ffffff;
}

@media (max-width:60em) {
  .emotion-92 {
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
    padding: 2rem 2rem 1rem;
  }

  .emotion-92 svg {
    -webkit-transform: scale(1.4);
    -ms-transform: scale(1.4);
    transform: scale(1.4);
  }
}

.emotion-96 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  opacity: 0.3;
  color: #ffffff;
  font-family: Helvetica,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
  margin: 2em auto 0;
  text-align: center;
  -webkit-letter-spacing: -0.1116279px;
  -moz-letter-spacing: -0.1116279px;
  -ms-letter-spacing: -0.1116279px;
  letter-spacing: -0.1116279px;
}

<div>
  <footer
    class="emotion-100 emotion-101"
  >
    <div
      class="emotion-98 emotion-99"
    >
      <div>
        <div
          class="emotion-88 emotion-89"
        >
          <ul
            class="emotion-20 emotion-21"
          >
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Home
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                About
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Events
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Cities
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Hotels
              </a>
            </li>
          </ul>
          <ul
            class="emotion-20 emotion-21"
          >
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Leagues
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                City Guides
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Tips
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                News
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Blog
              </a>
            </li>
          </ul>
          <ul
            class="emotion-20 emotion-21"
          >
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Contact
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Press
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Terms of Service
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Ticket Purchase Terms
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Privacy
              </a>
            </li>
          </ul>
          <ul
            class="emotion-20 emotion-21"
          >
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Our Guarantee
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                FAQ
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Sign-up / Login
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Download App for iOS
              </a>
            </li>
            <li
              class="emotion-2 emotion-3"
            >
              <a
                class="emotion-0 emotion-1"
              >
                Download App for Android
              </a>
            </li>
          </ul>
        </div>
        <div
          class="emotion-94 emotion-95"
        >
          <div
            class="emotion-90 emotion-91"
          >
            <a
              href="/"
            >
              <img
                src="mocked-file"
              />
            </a>
          </div>
          <div
            class="emotion-92 emotion-93"
          >
            <a
              href=""
            >
              <svg
                color="#d1d1d1"
                fill="#d1d1d1"
                height="34"
                width="16"
              >
                <path
                  d="M3.545 34V16.998H0v-5.859h3.545V7.622c0-4.779 2-7.622 7.688-7.622h4.734v5.86H13.01c-2.214 0-2.36.82-2.36 2.347l-.01 2.932H16l-.628 5.859h-4.734V34H3.545z"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
            <a
              href=""
            >
              <svg
                color="#d1d1d1"
                fill="#d1d1d1"
                height="32"
                width="32"
              >
                <g
                  fill-rule="evenodd"
                >
                  <rect
                    height="32"
                    rx="11"
                    width="32"
                  />
                  <circle
                    cx="24.686"
                    cy="6.4"
                    fill="#000"
                    r="2.743"
                  />
                  <circle
                    cx="16.457"
                    cy="16.457"
                    r="6.729"
                    stroke="#000"
                    stroke-width="3"
                  />
                </g>
              </svg>
            </a>
            <a
              href=""
            >
              <svg
                color="#d1d1d1"
                fill="#d1d1d1"
                height="29"
                width="34"
              >
                <path
                  d="M29.995 4.585A7.27 7.27 0 0033.06.536c-1.349.84-2.84 1.447-4.43 1.776C27.36.888 25.546 0 23.539 0c-3.852 0-6.973 3.278-6.973 7.32 0 .574.058 1.133.178 1.667C10.948 8.682 5.809 5.77 2.367 1.338a7.581 7.581 0 00-.944 3.682c0 2.539 1.231 4.78 3.103 6.095a6.75 6.75 0 01-3.161-.915v.09c0 3.548 2.403 6.508 5.598 7.178a6.523 6.523 0 01-1.84.257 6.59 6.59 0 01-1.311-.13c.886 2.908 3.462 5.027 6.516 5.084-2.39 1.965-5.397 3.133-8.664 3.133-.562 0-1.12-.031-1.664-.1C3.087 27.787 6.753 29 10.693 29c12.832 0 19.845-11.155 19.845-20.83 0-.318-.004-.634-.018-.946A14.536 14.536 0 0034 3.433a13.419 13.419 0 01-4.005 1.152z"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
            <a
              href=""
            >
              <svg
                color="#d1d1d1"
                fill="#d1d1d1"
                height="21"
                width="30"
              >
                <path
                  d="M30 15.846V5.154S30 0 24.83 0H5.168S0 0 0 5.154v10.692S0 21 5.168 21H24.83S30 21 30 15.846m-9.176-5.331l-9.82 5.75V4.763l9.82 5.752"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
            <a
              href=""
            >
              <svg
                color="#d1d1d1"
                fill="#d1d1d1"
                height="28"
                width="28"
              >
                <path
                  d="M.706 8.978h5.458v19.02H.706V8.978zM3.296 6.6h-.04C1.281 6.6 0 5.145 0 3.304 0 1.425 1.319 0 3.334 0c2.013 0 3.251 1.422 3.29 3.299 0 1.841-1.277 3.3-3.328 3.3zM28 28h-6.188v-9.844c0-2.577-.968-4.333-3.097-4.333-1.628 0-2.534 1.185-2.955 2.33-.158.409-.133.98-.133 1.555V28H9.496s.079-17.437 0-19.022h6.13v2.985c.363-1.305 2.322-3.168 5.448-3.168 3.879 0 6.926 2.738 6.926 8.627V28z"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <p
        class="emotion-96 emotion-97"
      >
        Some content licensed under a Creative Commons Attribution 3.0 License. SportsWhereIAm is a registered trademark of Where I Am Group © 2012-2021 WhereIam Group Pty Ltd.
      </p>
    </div>
  </footer>
</div>
`;
